### PR TITLE
fix: add try catch for response.json() to catch error on export

### PR DESCRIPTION
**Pull request by FannyGaudin**

### CODE REVIEW

File: src/kili/core/graphql/graphql_client.py [hunk 1]
```diff
@@ -224,6 +228,12 @@ def _get_kili_app_version(self) -> Optional[str]:
         """
         url = self.endpoint.replace("/graphql", "/version")
         response = self.http_client.get(url, timeout=30)
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as err:
+            self.logger.warning(f"An error occurred while processing the response: {err}")
+            raise RuntimeError(f"An error occurred while processing the response: {err}")
```
Comment (FannyGaudin) [inline src/kili/core/graphql/graphql_client.py:235]: I'm not sure about the RuntimeError maybe there is a better Error ?

File: src/kili/log/logging.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import logging
+import sys
+
+logger = logging.getLogger("kili")
```
Comment (Jonas1312) [inline src/kili/log/logging.py:4]: you can use this logger everywhere in the codebase

replace the several `getLogger( kili something, or __name__ )`  with the new global logger

https://github.com/search?q=repo%3Akili-technology%2Fkili-python-sdk%20getLogger&type=code

File: .github/scripts/upload_test_stats_datadog.py [hunk 3]
```diff
@@ -77,6 +79,11 @@ def get_workflow_runs_from_github() -> List[Dict]:
     while True:
         print("Fetching page", page, "...")
         response = requests.get(url + f"&page={page}", headers=HEADERS, timeout=30)
+        try:
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as err:
+            logger.exception("An error occurred while processing the response: %s", err)
+            raise RuntimeError(f"An error occurred while processing the response: {err}") from err
```
Comment (Jonas1312) [inline .github/scripts/upload_test_stats_datadog.py:86]:
```suggestion
            raise
```

you don't have to raise a custom error, logging + re-raising the requests error can be enough

but I'm fine with the runtimeerror too

File: src/kili/adapters/kili_api_gateway/asset/formatters.py [hunk 4]
```diff
@@ -3,18 +3,27 @@
...
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        logger.exception("An error occurred while processing the response: %s", err)
+        raise RuntimeError(f"An error occurred while processing the response: {err}") from err
     try:
-        return http_client.get(link, timeout=30).json()
+        return response.json()
     except json.JSONDecodeError:
         return {}
```
Comment (Jonas1312) [inline src/kili/adapters/kili_api_gateway/asset/formatters.py:28]: we should log here maybe, if the json can't be parsed (not sure why/when it happens...?)

**Review by Jonas1312 — COMMENTED** [review]
since the same code is repeated everywhere, we could create a dedicated function log_raise_for_status(requests.Response)

that calls response.raise_for_status() and logs if there's an error

File: src/kili/log/logging.py (hunk 2 above)
Comment (FannyGaudin) [reply src/kili/log/logging.py:4]: yes i'll done it in a second MR

**Review by Jonas1312 — APPROVED** [review]

**Review by aurelienlombard — APPROVED** [review]

**Pull request merged**